### PR TITLE
Make queue worker wait for setup before taking jobs

### PR DESCRIPTION
**Pull request by evilstreak**
When we moved the predictor into a subprocess, the call to `setup()` stopped blocking because the work isn't happening in the main thread.

Reuse the "prediction finished" communication channel to indicate that the setup is finished. Rename `PREDICTION_DONE` to `PROCESSING_DONE` to make it a little more generic.

I'm reluctant to add a test that has a minimum run time of 15s, but the timing on setup has a slightly non-deterministic amount of delay because of getting the docker container booted.

### CODE REVIEW

File: python/cog/server/runner.py [hunk 1]
```diff
@@ -51,13 +51,21 @@ def setup(self) -> None:
         self.predictor_process = multiprocessing.get_context("spawn").Process(
             target=self._start_predictor_process
         )
+
+        self._is_processing = True
         self.predictor_process.start()
 
+        while self.is_processing():
```
[the remark this answers is not in the document]
Comment (evilstreak) [reply python/cog/server/runner.py]: Good shout. I've added an infinite poll here. We might need to add some more logic to prevent failed setups from tying up the main process forever, but that's a problem with the current behaviour anyway I think 🤔

**Comment by bfirsh** [comment]
I wonder if this is a good excuse to add a class of test that only runs on main instead of every PR.
